Updated metadata documentation part 3 - probabilities

bjwheltor opened the pull request:
Description

An initial draft of the Probabilistic Information metadata document (a re-working of information currently hosted on an internal Met Office site). This creates a Probabilities.rst file.

brhooper commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+ 
+Cell methods can be used to describe any simple statistical method
+that has been apply to a variable.
+However, this leads to the question over over whether a statistical process
```
```suggestion
However, this leads to the question of whether a statistical process
```

bjwheltor replied:
Ta

brhooper commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+This is likely to depend on exactly who that user is
+and what they need to know about the data.
+Two extreme examples of how the cell methods might be presented
+for the same varaible (in this case a maximum temperature in a period)
```
```suggestion
for the same variable (in this case a maximum temperature in a period)
```

bjwheltor replied:
Think I need the spell checker plugin. Thanks.

brhooper commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+The other processing steps tell you more about how it was generated
+and are really acting as a substitute for provenance metadata.
+This can obscure the essential statistical information,
+making it harder to understand what the varaible actually represents.
```
```suggestion
making it harder to understand what the variable actually represents.
```

bjwheltor replied:
Changed

brhooper reviewed (changes requested):
Mostly this review contains corrections for typos and some re-phrasing suggestions to aid readability and/or clarity. 

I'm suggested that it might not be wholly accurate to title this section 'Probabilities' due to the range of different types of forecasts discussed. I've also suggested slightly changing the description of percentiles to more explicitly outline what a percentile is.

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+Context
+-------
+
+As IMPROVER is inherently probabalistic,
```
Corrected

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+-------
+
+As IMPROVER is inherently probabalistic,
+it seems appropriate to have a section specifically focussed
```
Changed.

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+
+As IMPROVER is inherently probabalistic,
+it seems appropriate to have a section specifically focussed
+on the represntation of probabalistic information in the metadata.
```
3 errors in the first sentence really isn't good! - corrected

bjwheltor commented on doc/source/index.rst:
```diff
@@ -24,6 +24,7 @@
 
    metadata/Introduction
    metadata/Principles
+   metadata/Probabilities
```
I wasn't completely happy with the way this section was structured anyway, so now that you've give my a prompt, I'm making some changes, namely:

 - Renaming it to "Probability Distribution"
 - Splitting out the cell methods part into a separate section "Statistical Processing"
 - This allows that background and context sections in to be merged with a shallower section hierarchy

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+Background
+**********
+
+Probabilistic information can be represented in three different ways: 
```
Removed this sentence and tied this into the representation of the probability distribution; hopefully this is more accurate.

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+
+* Ensemble members (realizations or scenarios) 
+* Probabilities of being above, below or between a set of thresholds 
+* Percentile values, representing the value of the diagnostics 
```
Whilst I agree my sentence is not really a definition, just a statement of what the data looks like from a technical perspective, I'm not quite comfortable with your definition (partly because is contains the word "observed", which can be interpreted in ab unintended way), so I've suggested "...representing thresholds of the distribution at or below which a given percentage of values fall."

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+Probabilistic information can be represented in three different ways: 
+
+* Ensemble members (realizations or scenarios) 
+* Probabilities of being above, below or between a set of thresholds 
```
Again, I've chosen to omit the word observed.
I've also made the ensemble members bullet more consistent by adding a similar description

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+Ensemble members
+****************
+
+This is the simplest form, as it is a natural extension to deterministic data,
```
Changed

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+
+This is probably the second most strightforward form,
+as again it still represents actual sets of values of the diagnostic,
+but instead of a set of realizations (consistent over time),
```
Good point

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+*************
+
+This is a more interesting form,
+as the diagnostics values are transformed into a set of probabilities,
```
Added text

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+    ``sum, sum_of_squares``,
+    ``variance``.
+
+If any method other than ``point`` is specified for a given axis,
```
Thanks, this communicates the use of point much more clearly

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+is unhelpful and potentially confusing within IMPROVER
+and should be omitted.
+
+There are two main ways in which cell methods is used
```
Changed.

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+
+2. Maximum, minimum and sum methods applied to time for probability values,
+   using a cell methods statement of the form below;
+   note that for the probablitie valuee, there is a non-standard comment
```
Strangely, this already seems to have been change to a slightly different wording (without the typos).

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,346 @@
+
+            air_temperature: cell_methods = "time: maximum (interval: 1 hour)" ;
+
+The non-standardised part is included in the same way,
```
It would be, but as there is only one standardised part, `interval`, which I'm trying to stop people using here, I didn't wantto. However, I've now added some text to point the reader to the CF Convention and re-structured to read a bit better (I hope).

bjwheltor commented:
@brhooper - Thanks for all your comments. I've now updated the document. Hopefully, all typos are corrected (but I may have introduced more). I've taken on some of your suggested changes with modifications, so see what you think. I've also split the section into 2 as I think this is clearer.

brhooper commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -1,42 +1,38 @@
-cell methods which are also heavily used within IMPROVER.
+As IMPROVER is inherently probabilistic,
+it seems appropriate to have a section specifically focused
+on the representation of probabality distributions in the metadata.
```
```suggestion
on the representation of probability distributions in the metadata.
```
Sorry, I must have missed this one on the first pass!

bjwheltor replied:
corrected.

brhooper commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -69,12 +65,14 @@ which runs every hour to generate a 3-member ensemble:
-but instead of a set of realizations (consistent over time),
-the set of percentile values represent the values at a set of probability levels.
+as again it still represents actual sets of values of the diagnostic.
+Instead of a set of realizations (consistent over time),
```
I'm still not quite sure what "consistent over time" means here in the context of realizations, is this something which could be clarified or elaborated upon, if it's important?

bjwheltor replied:
Thanks for flagging this up - I've tried to make this clearer by adding to the bracketed text, so it now reads "separate scenarios, each self-consistent over time"

brhooper commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -69,12 +65,14 @@ which runs every hour to generate a 3-member ensemble:
-the set of percentile values represent the values at a set of probability levels.
+as again it still represents actual sets of values of the diagnostic.
+Instead of a set of realizations (consistent over time),
+the set of percentile values represent the values of a set of thresholds
```
```suggestion
the set of percentiles represent thresholds
```
This is currently quite word-y, and the words "values" and "set" appear a few times each, which makes it more difficult to read

brhooper commented on doc/source/metadata/Statistics.rst:
```diff
@@ -0,0 +1,198 @@
+Context
+-------
+
+As IMPROVER both processes forecast data that has been
```
```suggestion
IMPROVER both processes forecast data that has been
```

bjwheltor replied:
Much better!

brhooper commented on doc/source/metadata/Statistics.rst:
```diff
@@ -0,0 +1,198 @@
+-------
+
+As IMPROVER both processes forecast data that has been
+statistically process end supports a range of statistical processing steps.
```
```suggestion
statistically processed and supports a range of statistical processing steps.
```

bjwheltor replied:
Good spot!

brhooper commented on doc/source/metadata/Statistics.rst:
```diff
@@ -0,0 +1,198 @@
+
+            air_temperature: cell_methods = "time: maximum (interval: 1 hour)" ;
+
+The non-standardised part is included in the same way
```
This section uses both the American-English (standardized) and UK-English (standardised) spellings. I think the American spelling for "realizations" has been used throughout the metadata documentation, so can the whole document be aligned on this please?

brhooper commented on doc/source/metadata/Statistics.rst:
```diff
@@ -0,0 +1,198 @@
+    air_temperature: cell_methods = "time: mean (time-weighted)" ;
+ 
+If both standardised and non-standardised parts are present,
+the latter must be proceeded by ``comment:`` keyword;
```
```suggestion
the latter must be preceded by the ``comment:`` keyword;
```

bjwheltor replied:
Corrected

brhooper commented on doc/source/metadata/Statistics.rst:
```diff
@@ -0,0 +1,198 @@
+for an example of this, see the `CF Metadata Conventions`_.
+
+Cell methods can be used to describe any simple statistical method
+that has been apply to a variable.
```
```suggestion
that has been applied to a variable.
```

bjwheltor replied:
Corrected.

brhooper commented on doc/source/metadata/Statistics.rst:
```diff
@@ -0,0 +1,198 @@
+
+The use of the ``interval`` within the extra information in cell methods
+is unhelpful and potentially confusing within IMPROVER
+and should be omitted.s
```
```suggestion
and should be omitted.
```

bjwheltor commented:
@brhooper - Thanks for the further corrections - I suspect I introduced more typos when i updated and split the text. Hopefully, I've addressed all these now.

brhooper commented on doc/source/metadata/Statistics.rst:
```diff
@@ -0,0 +1,198 @@
+-------
+
+IMPROVER both processes forecast data that has been
+statistically process and supports a range of statistical processing steps.
```
```suggestion
statistically processed and supports a range of statistical processing steps.
```
Missed this one

bjwheltor replied:
So did I - good spot.

brhooper commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,179 @@
+This included some extensions to the `CF Metadata Conventions`_ 
+which provide limited support in this area.
+
+Probabality distributions can be represented in one of three different ways: 
```
```suggestion
Probability distributions can be represented in one of three different ways: 
```

bjwheltor replied:
I really can't spell that word! - corrected.

brhooper reviewed (changes requested):
Sorry, 2 more typo corrections, then I'm happy to approve it

bjwheltor commented:
@brhooper - Thanks for (hopefully) the final few corrections - now addressed.

brhooper reviewed (approved):
Thanks for your changes @bjwheltor, sorry for going back-and-forth a couple of times for typos. Happy to approve this now.

gavinevans reviewed (changes requested):
Thanks @bjwheltor 👍 

I've added a few minor queries.

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,179 @@
+  a possible value of the diagnostic of interest;
+* Probabilities of the value of the diagnostic being above, below or between
+  a set of thresholds;
+* Percentile values, representing thresholds of the distribution of the
```
Done

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,179 @@
+		    percentile:long_name = "percentile" ;
+
+    data:
+     percentile = 5, 10, 15, 20, 30, 40, 50, 60, 70, 80, 85, 90, 95 ;0 ;
```
Definitely a typo - fixed.

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,179 @@
+* Dimension ``threshold``
+* Coordinate variable ``threshold``, with:
+
+  * Units appropriate to the original diagnostic
+    (indicated vy ``V`` in the following text) 
```
It's a typo - it should say "by  ``V``". I think I'll leave it in corrected form, as this is the first place that the 'original variable' is referred to, so it's important to make sure the reader knows that this is "V" in the example.

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,179 @@
+    * ``less_than`` 
+    * ``less_than_or_equal_to``
+
+An example would be a set of set of probabilities of temperature
```
Good spot - removed.

bjwheltor commented on doc/source/metadata/Statistics.rst:
```diff
@@ -0,0 +1,198 @@
+.. _stat-section:
+
+Statistical Processsing
```
Corrected.

bjwheltor commented:
@gavinevans - thanks for your review. I have addressed the areas that you've raised. I'm afraid I've also updated another section, which describes the default interpretation of cell_methods (this is in Statistical Processing in the Overview of Cell Methods section immediately after the description of "method"). I've done this, as, after re-reading the section in the CF Standard on this yesterday (to respond on a StaGE question), I realised that I had misinterpreted some aspects of cell_methods usage and that this section was misleading. I've tried to proof read this quite carefully, but obviously come back with any corrections that are required. Thanks.

gavinevans reviewed (changes requested):
Thanks for the update, @bjwheltor 👍 

I've pointed out a few very minor typos.

bjwheltor commented on doc/source/metadata/Statistics.rst:
```diff
@@ -0,0 +1,210 @@
+Precipitation rate is intensive; it would be described by
+``cell_methods = "time: point"`` and requires no time bounds.
+Precipitation accumulation is extensive; it would be described by
+``cell_methods = "time: sum"`` and does required time bounds.
```
Corrected.

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,179 @@
+As IMPROVER is inherently probabilistic,
+it seems appropriate to have a section specifically focused
+on the representation of probability distributions in the metadata.
+This included some extensions to the `CF Metadata Conventions`_ 
```
Yes, this should be includes - changed.

bjwheltor commented on doc/source/metadata/Probabilities.rst:
```diff
@@ -0,0 +1,179 @@
+Percentiles
+-----------
+
+This is probably the second most strightforward form,
```
Corrected

bjwheltor commented:
Thanks @gavinevans - I've made the further changes.
